grp/simple.gi

```diff
@@ -887,9 +887,101 @@ local id;
   return DataAboutSimpleGroup(id);
 end);
 
+# Tables for outer automorphisms from Bray/Holt/Roney-Dougal, p. 36/37
+BindGlobal("OuterAutoSimplePres",function(class,n,q)
+local p,e,f,rels,gp;
+  class:=UppercaseString(class);
+  p:=Factors(q)[1];
+  e:=LogInt(q,p);
+  if class="L" and n=2 then
+    f:=FreeGroup("d","f");
+    rels:=Concatenation("d",String(Gcd(q-1,2)),
+      "=f",String(e),"=[d,f]=1");
+  elif class="L" and n>=3 then
+    f:=FreeGroup("d","g","f");
+    rels:=Concatenation("d",String(Gcd(q-1,n)),
+      "=g2=f",String(e),"=[g,f]=1,d^g=D,d^f=d",String(p));
+  elif class="U" and n>=3 then
+    f:=FreeGroup("d","g","f");
+    rels:=Concatenation("d",String(Gcd(q+1,n)),
+      "=g2=1,f",String(e),"=g,d^g=D,d^f=d",String(p));
```

Comment:
This is not a request to make any changes, just a note in case you're not aware: there is a fairly-new function `StringFormatted` in GAP, which I now tend to use in situations like this, where I'm constructing a string from various parts. Depending on the context and the code in question, it can sometimes make the result more readable (and yes, it can sometimes make the result less readable).
```suggestion
    rels:=StringFormatted("d{}=g2=1, f{}=g, d^g=D, d^f=d{}", Gcd(q+1,n), e, p);
```

Reply:
Thank you -- indeed that will make it much nicer. I will change this later today once I'm done with classes.